lib/datadog/core/runtime/metrics.rb

```diff
@@ -78,6 +79,45 @@ def flush
               )
             end
           end
+
+          # Only on Ruby >= 3.2
+          try_flush do
+            if Core::Environment::YJIT.available?
```

Comment:
@GustavoCaso  I removed line 120 because it was mistakenly left over from a previous attempt at this.

I opted not to move the conditional check inline with the `end` to keep the style consistent with the rest of this file e.g. lines 57-81:

```ruby
          try_flush do
            if Core::Environment::VMCache.available?
              # Only on Ruby < 3.2
              gauge_if_not_nil(
                Core::Runtime::Ext::Metrics::METRIC_GLOBAL_CONSTANT_STATE,
                Core::Environment::VMCache.global_constant_state
              )
              ...
```